packages/cache/src/DiskLimiter.ts

```diff
@@ -0,0 +1,5 @@
+import pLimit from "p-limit";
+
+const MAX_FILE_OPERATIONS = 4000;
```

Comment:
Curious to know why we are using 4000?

Reply:
This is mostly arbitrary. Will experiment with this and iterate.
If perf is good and we don't hit the limit I will keep 4000, if we still hit the limit then I will reduce it, if it slows down things too much then I will try to increase it.